plasmapy/dispersion/__init__.py

```diff
@@ -0,0 +1,53 @@
+"""
+Tools for working with dispersion solvers.
+"""
+
+
+class DispersionInput:
+    """
+    Base class that stores the input to a dispersion solver.
+    """
+
+
+class DispersionOutput:
+    """
+    Base class that stores the output to a dispersion solver.
+
+    Parameters
+    ----------
+    input : `DispersionInput`
+        The input parameters passed to the dispersion solver that resulted in
+        this output.
+    k : array
+        Wavevectors.
+    omega : array
+        Angular frequencies. Can be complex, and must be the same size as *k*.
+    """
+    def __init__(self, input, omega, k):
+        if omega.size != k.size:
```

Comment:
I don't know much about dispersion solvers, but maybe `.shape` instead of `.size`? That's allow inputting 2D arrays with less bugs.